packages/runtime/src/next/client.ts

```diff
@@ -138,6 +138,27 @@ export type MakeswiftPageSnapshot = {
   cacheData: CacheData
 }
 
+const makeswiftComponentDocumentSchemaV2 = z.object({
+  id: z.string(),
+  name: z.string().nullable(),
+  locale: z.string(),
+  data: Schema.element,
+  type: z.string().nullable(),
+  siteId: z.string(),
+})
+
+export type MakeswiftComponentDocumentSchemaV2 = z.infer<typeof makeswiftComponentDocumentSchemaV2>
+
+const getMakeswiftComponentDocumentSchemaV2 = z.object({
+  elementTree: makeswiftComponentDocumentSchemaV2.nullable(),
+  parentLocaleElementTree: makeswiftComponentDocumentSchemaV2.nullable()
+})
+.and(z.union([
```

Comment:
Here I'm trying to enforce at least one of the two nullable fields being present. Basing this on a suggestion I saw here: https://github.com/colinhacks/zod/issues/61#issuecomment-1534255967

zod unions: https://zod.dev/?id=unions

Reply:
How come we're using both `.and` and `z.union`? It feels like we should use just one of those (I personally prefer `z.union`).

Also, is the very first case necessary? Seems like we're handling all valid cases within `z.union`. I see that this is how it's done in the suggestion you found. I'm wondering if it could be simplified a bit or if we have to do it that way